karyon3-eureka/src/main/java/com/netflix/karyon/eureka/KaryonHealthCheckHandler.java

```diff
@@ -0,0 +1,81 @@
+package com.netflix.karyon.eureka;
+
+import javax.inject.Singleton;
+
+import com.google.inject.Inject;
+import com.netflix.appinfo.HealthCheckHandler;
+import com.netflix.appinfo.InstanceInfo.InstanceStatus;
+import com.netflix.governator.LifecycleListener;
+import com.netflix.karyon.healthcheck.HealthCheck;
+import com.netflix.karyon.healthcheck.HealthChecks;
+
+/**
+ * Eureka HealthCheckHandler implementation that combines Karyon's HealthCheck with
+ * the injector lifecycle to auto-register with Eureka once the injector has been created
+ * and the HealthCheck is healthy.
+ * 
+ * @author elandau
+ */
+@Singleton
+public class KaryonHealthCheckHandler implements HealthCheckHandler, LifecycleListener {
```

Comment:
There is some slight behaviour difference between this healthcheckhandler and our current healthcheckhandler. The current handler does not invoke the health check until the local eureka status has changed away from STARTING (so the service need to, at some point, explicitly tell eureka that it is no longer in STARTING). What this means is that a server can still be STARTING even if Governator lifecycle is Started and all of it's components are healthy, which may be useful in some warm-up scenarios.